integration_tests/e2e/booking.cy.ts

```diff
@@ -0,0 +1,48 @@
+import premisesFactory from '../../server/testutils/factories/premises'
+import bookingFactory from '../../server/testutils/factories/booking'
+import BookingPage from '../pages/booking'
+import PremisesPage from '../pages/premises'
+import Page from '../pages/page'
+
+context('Booking', () => {
+  beforeEach(() => {
+    cy.task('reset')
+    cy.task('stubSignIn')
+    cy.task('stubAuthUser')
+  })
+
+  it('should show booking form', () => {
+    const booking = bookingFactory.build()
+    const premises = premisesFactory.buildList(5)
+    cy.task('stubBooking', booking)
+    cy.task('stubPremises', premises)
+
+    // Given I am signed in
+    cy.signIn()
+
+    // When I visit the booking page
+    const page = BookingPage.visit()
+
+    // Then I should see the booking form and enter details
+    page.getLabel('CRN')
+    page.getByInputByIdAndEnterDetails('CRN', 'A123456')
+
+    page.getLegend('What is the arrival date?')
+    page.arrivalDay().type('21')
+    page.arrivalMonth().type('07')
+    page.arrivalYear().type('22')
+
+    page.getLegend('What is the expected departure date?')
+    page.expectedDepartureDay().type('30')
+    page.expectedDepartureMonth().type('12')
+    page.expectedDepartureYear().type('22')
+
+    page.getLabel('Key Worker')
+    page.getByInputByIdAndEnterDetails('keyWorker', 'John Doe')
+
+    // And be able to click submit
+    page.clickSubmit()
+
+    Page.verifyOnPage(PremisesPage)
```

Comment:
This should be an expectation on its own, something like:

```
// Then I should be redirected to the premises page
```

Could we also set an expectation that the expected data gets passed to the API too?

Reply:
I think that would be wise. Do you suggest doing this via Wiremock?

Reply:
Yeah, I reckon so

Reply:
Should be able to use `getMatchingRequests`